linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -499,13 +529,14 @@ private final class IRChecker(unit: LinkingUnit, reporter: ErrorReporter) {
 
       case ArrayLength(array) =>
         typecheckExpr(array, env)
-        if (!array.tpe.isInstanceOf[ArrayType])
+        if (array.tpe != NullType && !array.tpe.isInstanceOf[ArrayType])
```

Comment:
I hadn't thought about this earlier, but shouldn't we also allow `NothingType` here?